src/main/java/org/jitsi/videobridge/AbstractEndpoint.java

```diff
@@ -354,8 +391,58 @@ public JSONObject getDebugState()
     public abstract void addRtpExtension(RtpExtension rtpExtension);
 
     /**
+     * @param newVideoAllocationPolicies
+     */
+    public abstract void setSenderVideoAllocationPolicies(
+        ImmutableMap<String, VideoAllocationPolicy> newVideoAllocationPolicies);
+
+    /**
+     * Notifies this instance that the max video constraints that the bridge
+     * needs to receive from this endpoint has changed. Each implementation
+     * handles this notification differently.
      *
-     * @param newVideoConstraints
+     * @param maxVideoConstraints the max video constraints that the bridge
+     *                            needs to receive from this endpoint
+     */
+    protected abstract void
+    maxReceiverVideoConstraintsChanged(VideoConstraints maxVideoConstraints);
+
+    /**
+     * The map of receiver endpoint id -> video constraints.
      */
-    public abstract void setVideoConstraints(Map<String, VideoConstraints> newVideoConstraints);
+    private final Map<String, VideoConstraints> receiverVideoConstraintsMap = new ConcurrentHashMap<>();
+
+    /**
+     * Notifies this instance that the specified endpoint wants to receive
+     * the specified video constraints from the endpoint attached to this
+     * instance (the sender).
+     *
+     * @param endpointId          the id that specifies the receiver endpoint
+     * @param newVideoConstraints the video constraints that the receiver
+     *                            wishes to receive.
+     */
+    public void addReceiver(String endpointId, VideoConstraints newVideoConstraints)
+    {
+        VideoConstraints oldVideoConstraints = receiverVideoConstraintsMap.put(endpointId, newVideoConstraints);
+        if (oldVideoConstraints == null
+            || !oldVideoConstraints.equals(newVideoConstraints))
+        {
+            receiverVideoConstraintsChanged(ImmutableMap.copyOf(receiverVideoConstraintsMap));
```

Comment:
I think in this particular case we should avoid the new map, because we're just calling a private method and it doesn't actually need a map. How about just passing `receiverVideoConstraintsMap.values()` (or `Collection.unmodifiableList(receiverVideoConstraintsMap.values())`)?